crates/provider/src/native.rs

```diff
@@ -296,19 +287,80 @@ impl<T: FileSystem + Send + Sync> Provider for NativeProvider<T> {
         Ok(format!("tail -f {}/{}.log", self.tmp_dir, name))
     }
 
+    // TODO: Add test
     async fn pause(&self, node_name: &str) -> Result<(), ProviderError> {
-        // TODO: impl using run_command
+        let process = self
+            .process_map
+            .get(node_name)
+            .ok_or(ProviderError::MissingNodeInfo(
+                node_name.to_owned(),
+                "process".into(),
+            ))?;
+
+        let _ = self
+            .run_command(
+                vec![format!("kill -STOP {}", process.pid)],
+                NativeRunCommandOptions {
+                    is_failure_allowed: true,
+                },
+            )
+            .await?;
         Ok(())
     }
 
+    // TODO: Add test
     async fn resume(&self, node_name: &str) -> Result<(), ProviderError> {
-        // TODO: impl using run_command
+        let process = self
+            .process_map
+            .get(node_name)
+            .ok_or(ProviderError::MissingNodeInfo(
+                node_name.to_owned(),
+                "process".into(),
+            ))?;
+
+        let _ = self
+            .run_command(
+                vec![format!("kill -CONT {}", process.pid)],
+                NativeRunCommandOptions {
+                    is_failure_allowed: true,
+                },
+            )
+            .await?;
         Ok(())
     }
 
-    async fn restart(&mut self, node_name: &str, after_secs: u16) -> Result<bool, ProviderError> {
-        // TODO: impl using run_command
-        // use &mut here since we mostly need to update the the pid/maps
+    // TODO: Add test
+    async fn restart(&mut self, node_name: &str, _after_secs: u16) -> Result<bool, ProviderError> {
+        let pid = self
+            .process_map
+            .get(node_name)
+            .ok_or(ProviderError::MissingNodeInfo(
+                node_name.to_owned(),
+                "process".into(),
+            ))?;
+        self.run_command(
+            [format!("kill -9 {:?}", pid)].to_vec(),
+            NativeRunCommandOptions {
+                is_failure_allowed: true,
+            },
+        )
+        .await?;
+
+        let Some(process) = self.process_map.get_mut(node_name) else {
+          return Err(ProviderError::MissingNodeInfo(
+                      node_name.to_owned(),
+                      "process".into(),));
+      };
```

Comment:
We can use the same approach with ok_or

```suggestion
        let process = self
            .process_map
            .get_mut(node_name)
            .ok_or(ProviderError::MissingNodeInfo(
                node_name.to_owned(),
                "process".into(),
            ))?;
```